### PR TITLE
Added support to specify token, host and node source at the submitJob pragma

### PR DESCRIPTION
**Pull request by carolinepacheco**

### CODE REVIEW

**Review by andrewssobral — APPROVED** [review]
OK for me

File: proactive-jupyter-kernel/kernel.py [hunk 1]
```diff
@@ -1431,6 +1434,62 @@ def __set_generic_information_from_input_data__(self, proactive_common, input_da
...
+            proactive_common.addGenericInformation("NODESOURCENAME", NODE_SOURCE_NAME)
+            from string import Template
+            selection_script_template = Template("""
+selected = false
+NODE_SOURCE_NAME = "$NODE_SOURCE_NAME"
+if (NODE_SOURCE_NAME) {
+    selected = (NODE_SOURCE_NAME.equals(System.getProperty("proactive.node.nodesource")));
+} else {
+    selected = true
+}
```
Comment (mklkun) [inline proactive-jupyter-kernel/kernel.py:1450]: Can the template scripts be saved in a different location than in kernel.py? 
Could we have resources files to store them?

File: proactive-jupyter-kernel/kernel.py [hunk 2]
```diff
@@ -1431,6 +1434,62 @@ def __set_generic_information_from_input_data__(self, proactive_common, input_da
...
+            HOST_NAME = input_data['host']
+            from string import Template
+            selection_script_template = Template("""
+selected = false
+HOST_NAME = "$HOST_NAME"
+if (HOST_NAME) {
+    selected = (nodehost.toLowerCase() == HOST_NAME)
+} else {
+    selected = true
+}
```
Comment (mklkun) [inline proactive-jupyter-kernel/kernel.py:1475]: Same

File: proactive-jupyter-kernel/kernel.py [hunk 3]
```diff
@@ -1431,6 +1434,62 @@ def __set_generic_information_from_input_data__(self, proactive_common, input_da
             for gen_info in input_data['generic_info']:
                 proactive_common.addGenericInformation(gen_info[0], gen_info[1])
 
+    def __set_job_nodesource__(self, proactive_common, input_data):
+        if 'nodesource' in input_data:
+            self.__kernel_print_ok_message__('Adding node source information ...\n')
+            NODE_SOURCE_NAME = input_data['nodesource']
+            proactive_common.addGenericInformation("NODESOURCENAME", NODE_SOURCE_NAME)
```
Comment (mklkun) [inline proactive-jupyter-kernel/kernel.py:1441]: Is this necessary? Generic information + selection script to set a target NS?

File: proactive-jupyter-kernel/kernel.py [hunk 4]
```diff
@@ -1962,6 +2021,53 @@ def __submit_job__(self, input_data):
         else:
             input_data['name'] = self.job_name
 
+        if self.added_nodesource_submitjob:
+            self.proactive_job.removeGenericInformation("NODESOURCENAME")
```
Comment (mklkun) [inline proactive-jupyter-kernel/kernel.py:2025]: Should we not test if the GI `NODESOURCENAME` is not yet set?

Comment (andrewssobral) [reply proactive-jupyter-kernel/kernel.py:2025]: We already do it. The variable self.added_nodesource_submitjob is set to True only if the nodesource is added using submitJob pragma.

File: proactive-jupyter-kernel/kernel.py (hunk 3 above)
Comment (andrewssobral) [reply proactive-jupyter-kernel/kernel.py:1441]: Yes.

File: proactive-jupyter-kernel/kernel.py (hunk 1 above)
Comment (andrewssobral) [reply proactive-jupyter-kernel/kernel.py:1450]: I am not sure, as this script is standard and not changed along the past 10 years, so this is why i put it in the code.

File: proactive-jupyter-kernel/kernel.py (hunk 4 above)
Comment (mklkun) [reply proactive-jupyter-kernel/kernel.py:2025]: Yes! I saw that and deleted the comment lol :p

File: proactive-jupyter-kernel/kernel.py (hunk 1 above)
Comment (andrewssobral) [reply proactive-jupyter-kernel/kernel.py:1450]: I know this is not a good practice, but we can refactor later. We can create a template folder to store those scripts as template and load them in the code. But we need to test this concept before to not break anything.

**Review by mklkun — APPROVED** [review]

**Pull request merged**